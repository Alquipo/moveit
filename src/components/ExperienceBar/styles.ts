import styled, { css } from 'styled-components'

export const Wrapper = styled.header`
  display: flex;
  align-items: center;

  span {
    font-size: 1rem;
  }
`
export const ExperienceBar = styled.div`
  ${({ theme }) => css`
    flex: 1;
    height: 4px;
    border-radius: 4px;
    background-color: ${theme.colors.grayLine};
    margin: 0 1.5rem;
    position: relative;
  `}
`
export const ExperienceBarProgress = styled.div`
  ${({ theme }) => css`
    height: 4px;
    border-radius: 4px;
    background-color: ${theme.colors.green};
  `}
`

export const CurrentExperience = styled.span`
  position: absolute;
  top: 12px;
  transform: translateX(-50%);
`
