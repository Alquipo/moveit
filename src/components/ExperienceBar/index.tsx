import { ChallengesContext } from 'contexts/ChallengesContext'
import { useContext } from 'react'

import * as S from './styles'

const ExperienceBar = () => {
  const { currentExperience, experienceToNextLevel } = useContext(
    ChallengesContext
  )

  const percentToNextLevel =
    Math.round(currentExperience * 100) / experienceToNextLevel

  return (
    <S.Wrapper>
      <span>0 xp</span>
      <S.ExperienceBar>
        <S.ExperienceBarProgress style={{ width: `${percentToNextLevel}%` }} />

        <S.CurrentExperience style={{ left: `${percentToNextLevel}%` }}>
          <b>^</b>
          {currentExperience === 0 ? null : `${currentExperience} xp`}
        </S.CurrentExperience>
      </S.ExperienceBar>

      <span>{experienceToNextLevel} xp</span>
    </S.Wrapper>
  )
}

export default ExperienceBar
