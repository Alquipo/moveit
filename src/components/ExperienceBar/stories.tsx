import { Story, Meta } from '@storybook/react/types-6-0'
import ExperienceBar from '.'

export default {
  title: 'ExperienceBar',
  component: ExperienceBar
} as Meta

export const Default: Story = (args) => <ExperienceBar {...args} />
