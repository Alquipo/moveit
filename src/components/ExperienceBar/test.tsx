import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import ExperienceBar from '.'

describe('<ExperienceBar />', () => {
  it('should render the ExperienceBar correctly', () => {
    const { container } = renderWithTheme(<ExperienceBar />)

    expect(screen.getByText('0 xp')).toBeInTheDocument()
    expect(screen.getByText('300 xp')).toBeInTheDocument()
    expect(screen.getByText('600 xp')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
