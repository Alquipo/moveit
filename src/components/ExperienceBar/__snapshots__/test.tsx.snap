// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExperienceBar /> should render the ExperienceBar correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #EEEEEE;
  -webkit-transition: color 0.3s ease-in-out;
  transition: color 0.3s ease-in-out;
}

.c0 span {
  font-size: 1rem;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 4px;
  border-radius: 4px;
  background-color: #DCDDE0;
  margin: 0 1.5rem;
  position: relative;
}

.c2 {
  height: 4px;
  border-radius: 4px;
  background-color: #04d361;
}

.c3 {
  position: absolute;
  top: 12px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

<div
  class="c0"
>
  <span>
    0 xp
  </span>
  <div
    class="c1"
  >
    <div
      class="c2"
      style="width: 50%;"
    />
    <span
      class="c3"
      style="left: 50%;"
    >
      300 xp
    </span>
  </div>
  <span>
    600 xp
  </span>
</div>
`;
